pkg/virt-controller/watch/vm.go

```diff
@@ -879,6 +880,35 @@ func (c *VMController) handleVolumeRequests(vm *virtv1.VirtualMachine, vmi *virt
 	return nil
 }
 
+func (c *VMController) handleVolumeUpdateRequest(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) error {
+	if !c.clusterConfig.VolumesUpdateStrategyEnabled() {
+		return nil
+	}
+	if vmi == nil {
+		return nil
+	}
+	if equality.Semantic.DeepEqual(vmi.Spec.Volumes, vm.Spec.Template.Spec.Volumes) {
+		return nil
+	}
+	vmConditions := controller.NewVirtualMachineConditionManager()
+	switch {
+	case vm.Spec.UpdateVolumesStrategy == nil ||
+		*vm.Spec.UpdateVolumesStrategy == virtv1.UpdateVolumesStrategyReplacement:
+		if !vmConditions.HasCondition(vm, virtv1.VirtualMachineRestartRequired) {
+			vmConditions.UpdateCondition(vm, &virtv1.VirtualMachineCondition{
+				Type:               virtv1.VirtualMachineRestartRequired,
+				LastTransitionTime: v1.Now(),
+				Status:             k8score.ConditionTrue,
+				Message:            "the volumes replacement is effective only after restart",
```

Comment:
If the condition is already set, this message is lost. Could that be an issue?

Reply:
Right, it, however, means that the VM needs already to be rebooted since there was already a change. The only way would be to append the message to the previous one. Not sure if this is really elegant. WDYT?

Reply:
Additionally. It is already the case for CPU and memory hotplug. If there is already the requiredRestart condition set, then we don't evaluate the live-updateable fields. See [here](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm.go#L3070)

Reply:
ACK, I think it's okay not to touch the message if the condition is already there.